packages/angular_devkit/build_angular/src/webpack/plugins/devtools-ignore-plugin.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Compilation, Compiler } from 'webpack';
+
+// Following the naming conventions from
+// https://sourcemaps.info/spec.html#h.ghqpj1ytqjbm
+const IGNORE_LIST = 'x_google_devtoolsIgnore';
+
+const PLUGIN_NAME = 'devtools-ignore-plugin';
+
+interface Options {
+  vendors?: false | { test: RegExp };
+  runtime?: false | { test: RegExp };
+}
+/**
+ * This plugin adds a field to source maps that identifies which sources are
+ * vendored or runtime-injected (aka third-party) sources. These are consumed by
+ * Chrome DevTools to automatically ignore-list sources.
+ */
+export class DevToolsIgnorePlugin {
+  constructor(private options: Options) {}
+
+  apply(compiler: Compiler) {
+    const { SourceMapSource } = compiler.webpack.sources;
+
+    compiler.hooks.compilation.tap(PLUGIN_NAME, (compilation) => {
+      compilation.hooks.processAssets.tap(
+        {
+          name: PLUGIN_NAME,
+          stage: Compilation.PROCESS_ASSETS_STAGE_DEV_TOOLING,
+          additionalAssets: true,
```

Comment:
Out of curiosity, do we need the `additionalAssets` option?

Reply:
Yes, but I won't be able to tell you exactly why. In my testing, the source map assets are received only when this is true.

Reply:
I'm referring to the assets for which `asset.map()` actually returns something.

Reply:
Oh wait, no, this was backwards. Without additional assets, the following are processed:
```
main.js
polyfills.js
styles.css
runtime.js
vendor.js
```

With additional assets, the following extra are processed:
```
main.js.map
polyfills.js.map
styles.css.map
runtime.js.map
vendor.js.map
```

So the plugin should also work _without_ `additionalAssets: true`.

Reply:
Removed this, works as expected.